File: programmers/짝지어제거하기.js
function solution(s) {
    const stack = []
    stack.push(s[0])
    for(let i = 1; i < s.length; i++) {
        const lastChar = stack[stack.length - 1]
        const currChar = s[i]
        if(lastChar === currChar) {
            stack.pop()
        } else {
            stack.push(currChar)
        }
    }
    return stack.length === 0 ? 1 : 0
}

console.log(solution('baabaa'))
console.log(solution('cdcd'))

File: programmers/짝지어제거하기.py
def solution(string):
    stack = [string[0]]
    for i in range(1, len(string)):
        currChar = string[i]
        if len(stack) == 0:
            stack.append(currChar)
            continue
        prevChar = stack[-1]
        if prevChar == currChar:
            stack.pop()
        else:
            stack.append(currChar)
    if len(stack) == 0:
        return 1
    else:
        return 0

def solution2(string): #첫번째 요소도 일관성유지
    stack = []
    for i in range(len(string)):
        currChar = string[i]
        if len(stack) == 0:
            stack.append(currChar)
            continue
        prevChar = stack[-1]
        if prevChar == currChar:
            stack.pop()
        else:
            stack.append(currChar)
    if len(stack) == 0:
        return 1
    else:
        return 0        

print(solution2('baabaa'))
print(solution2('cdcd'))